probe-rs/src/probe/ti_icdi/mod.rs

```diff
@@ -0,0 +1,289 @@
+mod gdb_interface;
+mod receive_buffer;
+mod usb_interface;
+
+use crate::{
+    CoreRegisterAddress, DebugProbe, DebugProbeError, DebugProbeSelector, Error, Memory, Probe,
+    WireProtocol,
+};
+
+use crate::architecture::arm::ap::{AccessPort, GenericAp, MemoryAp};
+use crate::architecture::arm::dp::DebugPortVersion;
+use crate::architecture::arm::memory::Component;
+use crate::architecture::arm::{
+    ApInformation, ArmChipInfo, ArmProbeInterface, DapAccess, MemoryApInformation, SwoAccess,
+    SwoConfig,
+};
+
+pub use usb_interface::list_icdi_devices;
+use usb_interface::IcdiUsbInterface;
+
+use crate::architecture::arm::memory::adi_v5_memory_interface::ArmProbe;
+use crate::probe::ti_icdi::gdb_interface::GdbRemoteInterface;
+use crate::Error as ProbeRsError;
+
+#[derive(Debug)]
+pub struct IcdiProbe {
+    device: IcdiUsbInterface,
+    protocol: WireProtocol,
+    name: String,
+    speed_setting: u8,
+}
+
+impl IcdiProbe {
+    pub fn as_memory(&mut self) -> Memory<'_> {
+        Memory::new(self, MemoryAp::new(0))
+    }
+}
+
+impl DebugProbe for IcdiProbe {
+    fn new_from_selector(
+        selector: impl Into<DebugProbeSelector>,
+    ) -> Result<Box<Self>, DebugProbeError>
+    where
+        Self: Sized,
+    {
+        let mut device = IcdiUsbInterface::new_from_selector(selector)?;
+        let ver = device.query_icdi_version()?;
+        let name = format!("ICDI S/N: {}, ver: {}", &device.serial_number, ver);
+        Ok(Box::new(Self {
+            device,
+            protocol: WireProtocol::Jtag,
+            name,
+            speed_setting: 0,
+        }))
+    }
+
+    fn get_name(&self) -> &str {
+        &self.name
+    }
+
+    fn speed(&self) -> u32 {
+        2000
```

Comment:
Do we not have a way to get back the speed? Either way I do not think hardcoding is the way.